rust/lance/src/dataset.rs

```diff
@@ -7798,4 +7798,148 @@ mod tests {
             values
         )
     }
+
+    #[tokio::test]
+    async fn test_geo_types() {
+        use geo_types::{coord, line_string, Rect};
+        use geoarrow::array::{LineStringBuilder, PointBuilder, PolygonBuilder};
+        use geoarrow::datatypes::{
+            Dimension, GeoArrowType, LineStringType, PointType, PolygonType,
+        };
+        use geoarrow_array::GeoArrowArray;
+
+        // 1. Creates arrow table with spatial data.
+        let point_type = PointType::new(Dimension::XY, Default::default());
+        let line_string_type = LineStringType::new(Dimension::XY, Default::default());
+        let polygon_type = PolygonType::new(Dimension::XY, Default::default());
+
+        let schema = arrow_schema::Schema::new(vec![
+            GeoArrowType::from(point_type.clone()).to_field("point", true),
```

Comment:
FYI there's a `to_field` [method on `PointType`](https://docs.rs/geoarrow-schema/latest/geoarrow_schema/struct.PointType.html#method.to_field), so you don't have to convert to `GeoArrowType`.